daemon/api_debug.go

```diff
@@ -63,18 +63,43 @@ func getBaseDeclaration(st *state.State) Response {
 	}, nil)
 }
 
+func checkConnectivity(theStore snapstate.StoreService) Response {
+	checkResult, err := theStore.ConnectivityCheck()
+	if err != nil {
+		return InternalError("cannot run connectivity check: %v", err)
+	}
+	status := ConnectivityStatus{Connectivity: true}
+	for host, reachable := range checkResult {
+		if !reachable {
+			status.Connectivity = false
+			status.Unreachable = append(status.Unreachable, host)
+		}
+	}
+	sort.Strings(status.Unreachable)
+
+	return SyncResponse(status, nil)
+}
+
 func getDebug(c *Command, r *http.Request, user *auth.UserState) Response {
 	query := r.URL.Query()
 	aspect := query.Get("aspect")
-	if aspect != "base-declaration" {
+	st := c.d.overlord.State()
+	switch aspect {
+	case "base-declaration":
+		st.Lock()
```

Comment:
Perhaps consider moving locking ahead of the switch statement. We can move it inside the methods if there is a need for it.